UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -314,6 +323,12 @@ void OnEnableHelper(Academy academy)
         {
             m_Info = new AgentInfo();
             m_Action = new AgentAction();
+            m_Sensors = new List<ISensor>();
+
+            // TODO deterministic sorting
+            var attachedSensors = GetComponents<SensorBase>();
```

Comment:
could init this with the capacity of of `atachedSensors` and add in a loop to avoid extra GC allocations.  But no big deal

Reply:
Yeah, I think you can also do `GetComponents<SensorBase>(m_Sensors)` too. I'll probably break this out to a virtual method so users can hook in their own ISensors too.

Reply:
I now do `m_Sensors.Capacity += attachedSensorComponents.Length;` before appending the ISensors